File: 452.用最少数量的箭引爆气球/452.用最少数量的箭引爆气球.cpp
/*
 * @Descroption: LeetCode 452. 用最少数量的箭引爆气球
 * @Author: EmoryHuang
 * @Date: 2021-08-12 20:01:27
 * @Method:
 * 贪心
 * 按照右端点排序，
 * 将左端点与之前引爆的值作比较，找到当前箭能引爆的最大数量
 */

class Solution {
   public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](vector<int> a, vector<int> b) { return a[1] < b[1]; });
        int pre = points[0][1];
        int ans = 1;
        for (auto point : points) {
            if (point[0] > pre) {
                ans++;
                pre = point[1];
            }
        }
        return ans;
    }
};

File: Solution/452.用最少数量的箭引爆气球/452.用最少数量的箭引爆气球.java
/*
 * @Descroption: LeetCode 452. 用最少数量的箭引爆气球
 * @Author: EmoryHuang
 * @Date: 2023-07-22 10:29:31
 * @Method:
 * 贪心，根据右端点从小到大排序
 */

import java.util.*;

class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (o1, o2) -> o1[1] < o2[1] ? -1 : 1);
        int pre = points[0][1], ans = 1;
        for (int i = 1; i < points.length; i++) {
            if (pre < points[i][0]) {
                ans++;
                pre = points[i][1];
            }
        }
        return ans;
    }
}